### ISSUE TITLE
MTE KASAN

### ISSUE THREAD

**Issue by W-M-R** [google/sanitizers#1823]
I would like to ask, where is the location of the arm64-specific mte in armv8a label storage, and the register that records the base address is wage? Thank you, looking forward to the reply of experts

**Comment by fmayer** [comment]
The tag storage is the same for kernel and userspace MTE. It is transparent to the kernel and is set up by the bootloader, and then hidden from the kernel.

What are you trying to achieve?

**Comment by W-M-R** [comment]
Thank you.I want to know where the MTE label is stored. I see that in Linux, qemu stores the label at addr >> shift, which is implemented by qemu. I want to know how the label storage location is implemented for CPUs that already have the MTE function. Is this determined by the chip manufacturer?

**Comment by fmayer** [comment]
Yes, that is a hardware detail. See https://developer.arm.com/documentation/ka005620/latest/ for instance

**Comment by W-M-R** [comment]
OK, thank you very much
